src/TTImp/ProcessBuiltin.idr

```diff
@@ -0,0 +1,163 @@
+||| Checking a %builtin pragma is correct.
+-- If we get more builtins it might be wise to move each builtin to it's own file.
+module TTImp.ProcessBuiltin
+
+import Libraries.Data.Bool.Extra
+import Libraries.Data.NameMap
+import Data.List
+
+import Core.Core
+import Core.Context
+import Core.Context.Log
+import Core.Env
+import Core.Metadata
+import Core.TT
+import Core.UnifyState
+
+import TTImp.TTImp
+
+||| Get the return type.
+getRetTy : {vars : _} -> Term vars -> (vars ** Term vars)
+getRetTy (Bind _ _ _ tm) = getRetTy tm
+getRetTy tm = (vars ** tm)
+
+||| Get the first non-erased argument type.
+getFirstNETy : {vars : _} -> Term vars -> (vars ** Term vars)
+getFirstNETy (Bind _ _ b tm) = if isErased $ multiplicity b
```

Comment:
Same comment here about the different types of binders: you only want to return
the binder if it's not erased *and* it's a `Pi`.